### PR TITLE
[Improve][ST-Engine] Improve job restart of all node down

### PR DESCRIPTION
**Pull request by ic4y**
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/PhysicalVertex.java [hunk 1]
```diff
@@ -374,16 +382,20 @@ public void cancel() {
...
         } else if (updateTaskState(ExecutionState.RUNNING, ExecutionState.CANCELING)) {
-            noticeTaskExecutionServiceCancel(Integer.MAX_VALUE);
+            noticeTaskExecutionServiceCancel();
         }
     }
 
     @SuppressWarnings("checkstyle:MagicNumber")
-    private void noticeTaskExecutionServiceCancel(int tryTimes) {
+    private void noticeTaskExecutionServiceCancel() {
+        //Check whether the node exists, and whether the Task on the node exists. If there is no direct update state
```
Comment (Grypse) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/PhysicalVertex.java:391]: Update comment to this `Check whether the node exists, and check whether any task is running on the node. If not, update the state directly`

**Comment by Grypse** [comment]
@TyrantLucifer PTAL. this solved e2e error.

**Comment by TyrantLucifer** [comment]
> @TyrantLucifer PTAL. this solved e2e error.

cc @EricJoy2048

**Review by EricJoy2048 — APPROVED** [review]
LGTM

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**